src/datumaro/plugins/framework_converter.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Callable, Optional
+
+import numpy as np
+
+from datumaro.components.annotation import AnnotationType
+from datumaro.components.dataset import Dataset
+
+TASK_ANN_TYPE = {
+    "classification": AnnotationType.label,
+    "detection": AnnotationType.bbox,
+    "instance_segmentation": AnnotationType.polygon,
+    "semantic_segmentation": AnnotationType.mask,
+}
+
+
+class FrameworkConverterFactory:
+    @staticmethod
+    def create_converter(framework):
+        if framework == "torch":
+            return DmTorchDataset
+        elif framework == "tf":
+            return DmTfDataset
+        else:
+            raise ValueError("Unsupported framework")
+
+
+class FrameworkConverter:
+    def __init__(self, dataset, subset, task):
+        self._dataset = dataset
+        self._subset = subset
+        self._task = task
+
+    def to_framework(self, framework, **kwargs):
+        converter_cls = FrameworkConverterFactory.create_converter(framework)
+        return converter_cls(
+            dataset=self._dataset,
+            subset=self._subset,
+            task=self._task,
+            **kwargs,
+        )
+
+
+try:
+    import torch
+
+    class DmTorchDataset(torch.utils.data.Dataset):
+        def __init__(
+            self,
+            dataset: Dataset,
+            subset: str,
+            task: str,
+            transform: Optional[Callable] = None,
+            target_transform: Optional[Callable] = None,
+        ):
+            self.dataset = dataset.get_subset(subset)
+            self.subset = subset
+            self.task = task
+            self.transform = transform
+            self.target_transform = target_transform
+            self._ids = []
+            for item in self.dataset:
+                self._ids.append(item.id)
+
+        def __len__(self) -> int:
+            return len(self.dataset)
+
+        def __getitem__(self, idx):
+            dataitem = self.dataset.get(id=self._ids[idx], subset=self.subset)
+            image = dataitem.media.data
+            if image.dtype == np.uint8 or image.max() > 1:
+                image = image.astype(np.float32) / 255
+
+            if len(image.shape) == 2:
+                image = np.expand_dims(image, axis=-1)
+
+            if self.task == "classification":
+                label = [
+                    ann.label
+                    for ann in dataitem.annotations
+                    if ann.type == TASK_ANN_TYPE[self.task]
+                ][0]
```

Comment:
Shouldn't we limit it from `classification` to `multi-class-classification` because we pick only one label. It is also preferred to add exception or an error log when there are multi labels.

Reply:
Thanks, I added log for warning this and new multilabel classification task at aa2110a077f18bcaa22ff07fef62e225958e9c4f.